### PR TITLE
Update java-based stacks to use EAP-7.3.x and plugin-java8-rhel8 images

### PR DESCRIPTION
**Pull request by svor**
Signed-off-by: svor <vsvydenk@redhat.com>

### What does this PR do?
Updates java-based stacks with new java images

- Fuse: plugin-java8-rhel8
- thorntail: plugin-java8-rhel8
- java-maven: plugin-java11-rhel8
- springboot: plugin-java8-rhel8
- vertx: plugin-java11-rhel8
- java-eap-maven: eap73-openjdk11-openshift-rhel8  **(run command doesn't work)**
 
### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-871

### CODE REVIEW

File: dependencies/che-devfile-registry/devfiles/02_java-jboss-fuse/meta.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 displayName: Red Hat Fuse
-description: Red Hat Fuse stack with EAP 7.2.7, OpenJDK 1.8 and Maven 3.5.4
+description: Red Hat Fuse stack with EAP 7.3.1, OpenJDK 1.8 and Maven 3.5.0
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/devfiles/02_java-jboss-fuse/meta.yaml:3]: that's weird that they went backwards with the version.

Also ... the new `deployment` container is at mvn 3.6... so will this be confusing/break stuff?

File: dependencies/che-devfile-registry/devfiles/02_java-jboss-fuse/devfile.yaml [hunk 2]
```diff
@@ -29,17 +38,15 @@ components:
     # TODO: switch to registry.redhat.io/jboss-eap-7/eap-xp1-openjdk8-openshift-rhel7:1.0 when it's live (June 1?)
     image: registry-proxy.engineering.redhat.com/rh-osbs/jboss-eap-7-eap-xp1-openjdk8-openshift-rhel7:latest
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/devfiles/02_java-jboss-fuse/devfile.yaml:39]: if we can switch to `registry.redhat.io/ubi8/openjdk-8:1.3` then we'll be on mvn 3.6.2-4 and jvn 1.8.0_252.

File: dependencies/che-devfile-registry/devfiles/02_java-jboss-fuse/meta.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
 tags: ["RHEL8", "Java", "OpenJDK", "Maven", "EAP", "Red Hat Fuse"]
```
Comment (nickboldt) [inline dependencies/che-devfile-registry/devfiles/02_java-jboss-fuse/meta.yaml:4]: if we switch to a JDK-only image (no EAP) we have to remember to update this too.

**Comment by svor** [comment]
still working on this PR, pls don't merge

**Pull request merged**